javascript/javascript2/week3/fastestPresser/fastestPresser.js

```diff
@@ -0,0 +1,101 @@
+let inputElement = document.getElementById("myInput");
+let startButton = document.querySelector(".start-button");
+let outputElementS = document.querySelector(".letterS");
+let outputElementL = document.querySelector(".letterL");
+
+const keypressCount = {
+    sCount: 0,
+    lCount: 0
+};
+
+let keyPressed = false; // check if any key was pressed
+
+// Add the keydown event listener when the page loads
+document.addEventListener("keydown", getKeydownCount);
+
+// Start button click event
+startButton.addEventListener("click", () => {
+    let runTime = inputElement.value;
+
+    // Check if the timer is set
+    if (!runTime) {
+        alert("Please set the timer!");
+        return;
+    }
+
+    keyPressed = false; // set the flag to false before starting the game
+    runTimeout(runTime);
+});
+
+let timeoutId;
+
+function runTimeout(runTime) {
+    timeoutId = setTimeout(() => {
+        if (!keyPressed) {
+            alert("No key was pressed until timeout!");
+            return;
+        }
+        anotherFunction();
+    }, runTime * 1000);
+}
+
+function getKeydownCount(event) {
+    if (event.key === "s") {
+        outputElementS.innerHTML = keypressCount.sCount++;
+    } else if (event.key === "l") {
+        outputElementL.innerHTML = keypressCount.lCount++;
+    }
+
+    keyPressed = true;
+}
+
+function anotherFunction() {
+    let inputValue1 = Number(outputElementS.innerHTML);
+    let inputValue2 = Number(outputElementL.innerHTML);
+
+    let maxValue = Math.max(inputValue1, inputValue2);
+    let maxValueElement;
+
+    if (inputValue1 === inputValue2) {
+        alert("Tied Score");
+        return;
+
+    } else {
+        if (maxValue === inputValue1) {
+            maxValueElement = outputElementS;
+        } else {
+            maxValueElement = outputElementL;
+        }
+        let parenting = maxValueElement.parentElement.parentElement;
+        parenting.style.backgroundColor = "red";
+    }
+}
+
+
+// Sop the logging and clear the setTimeout
+let stopButton = document.querySelector(".stop-button");
+
+stopButton.addEventListener("click", () => {
+    if (!inputElement.value) {
+        alert("Please set the timer & start the game!");
+        return;
+    } else {
+        stopTimer();
+    }
+});
+
+function stopTimer() {
+    clearTimeout(timeoutId);
+    logging = false;
+    anotherFunction();
+}
+
+
+// Reset button for reloading the page
+let resetButton = document.querySelector(".reset-button");
+resetButton.addEventListener("click", () => {
+    location.reload();
+});
+
+
```

Comment:
Nice work, I think need to add a function to prevent pressing the buttons/ending the game after the game is done.  you can take a look on this code it may inspire you 
```
`const startButton = document.getElementById("start-button");
const durationInput = document.getElementById("duration");
const feedbackDiv = document.getElementById("feedback");
let sCounter = 0;
let lCounter = 0;

startButton.addEventListener("click", startGame);

function startGame() {
  // Disable the input and start button
  durationInput.disabled = true;
  startButton.disabled = true;

  // Get the game duration
  const duration = durationInput.value;

  // Set a timeout for the specified duration
  setTimeout(endGame, duration * 1000);

  // Add an event listener for key presses
  document.addEventListener("keypress", handleKeyPress);
}

function handleKeyPress(event) {
  if (event.key === "s") {
    sCounter++;
  } else if (event.key === "l") {
    lCounter++;
  }
}

function endGame() {
  // Remove the event listener
  document.removeEventListener("keypress", handleKeyPress);

  // Determine the winner
  let winner;
  if (sCounter > lCounter) {
    winner = "S";
  } else if (lCounter > sCounter) {
    winner = "L";
  } else {
    winner = "It's a tie!";
  }

  // Show the results in the feedback div
  feedbackDiv.innerHTML = `The game is over! ${winner} wins with ${Math.max(sCounter, lCounter)} key presses.`;
} `
```